runtime/precompiles/src/lib.rs

```diff
@@ -58,53 +41,20 @@ impl Precompile for DeadbeefPrecompiled {
 			))?;
 		result_vec.extend(input.to_vec());
 
-		Ok((pallet_evm::ExitSucceed::Returned, result_vec, cost))
-	}
-}
-
-type PrecompiledCallable = fn(&[u8], Option<usize>)
-	-> core::result::Result<(pallet_evm::ExitSucceed, Vec<u8>, usize), pallet_evm::ExitError>;
-
-fn get_precompiled_func_from_address(address: &H160) -> Option<PrecompiledCallable> {
-	use core::str::FromStr;
-
-	// Note that addresses from_str should not start with 0x, just the hex value
-	let addr_deadbeef = H160::from_str("0000000000000000000000000000000000001000")
-		.expect("Invalid address at precompiles generation");
-
-	if *address == addr_deadbeef {
-		return Some(DeadbeefPrecompiled::execute);
-	}
-
-	None
-}
-
-impl Precompiles for ExperimentalMoonbeamPrecompiles {
-	fn execute(
-		address: H160,
-		input: &[u8],
-		target_gas: Option<usize>
-	) -> Option<
-		core::result::Result<
-			(pallet_evm::ExitSucceed, Vec<u8>, usize),
-			pallet_evm::ExitError,
-		>
-	> {
-		match get_precompiled_func_from_address(&address) {
-			Some(func) => return Some(func(input, target_gas)),
-			_ => {},
-		};
-
-		None
+		Ok((pallet_evm::ExitSucceed::Returned, result_vec))
 	}
 }
 
-pub type MoonbeamPrecompiles =
+pub type MoonbeamPrecompiles<Runtime> =
 (
-	pallet_evm::precompiles::ECRecover,
-	pallet_evm::precompiles::Sha256,
-	pallet_evm::precompiles::Ripemd160,
-	pallet_evm::precompiles::Identity,
+	pallet_evm_precompile_simple::ECRecover,
+	pallet_evm_precompile_simple::Sha256,
+	pallet_evm_precompile_simple::Ripemd160,
+	pallet_evm_precompile_simple::Identity,
+	// TODO Should we add blake2 or others from Frontier?
+	// What all precompiles do we want?
+	// https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go
```

Comment:
We were wondering what precompiles we wanted. When you mention the eight default, do you mean these ones? https://github.com/ethereum/go-ethereum/blob/3c46f5570bd674cf49c0113352ff79a4e026a5b8/core/vm/contracts.go#L56-L65

We can definitely add them.

Reply:
Yes those are the ones, the initial precompiles were added to substrate but since the evm has moved to frontier, it is probably there now